File: csharp/pangram/Pangram.cs
using System;
using System.Linq;

public static class Pangram {
    public static bool IsPangram(string input)
    {
        const char[26] Alphabet = StringComparer 'a' to 'z';
        var seen = new bool[26];
        foreach (var letter in input.ToLower())
        {
            if (char.IsLetter(letter))
            {
                seen[letter - 'a'] = true;
            }
        }
        return seen.All(b => b);
    }
}

File: c/pangram/src/pangram.c
#include "pangram.h"

bool is_pangram(char const* s) {
  unsigned long long bits = 0ull;
  unsigned char c;
  while (s && (c = *s++)) {
    if      ('Z' >= c && c >= 'A') { bits |= 1ull << (c - 'A'); }
    else if ('z' >= c && c >= 'a') { bits |= 1ull << (c - 'a'); }
  }
  return bits == 0x3FFFFFF;
}
